hyper_search.py

```diff
@@ -0,0 +1,126 @@
+
+
+from sklearn.model_selection import train_test_split
+import numpy as np
+
+def max_depth_search(model, X, y, cycles=20, depth_list=None, verbose=False):
+    
+    if depth_list is None:
+        depth_list = [1, 3, 6, 10, None]
```

Comment:
Set this as a default argument instead of setting the default to None and then also having an extra if statement